site/_posts/2020-07-20-breached-password-detection.md

```diff
@@ -0,0 +1,148 @@
+
+Ouch.
+
+## How FusionAuth detects breached passwords
+
+*Please note that this is a paid edition feature. You can [learn more about paid editions here](/pricing).*
+
+We have built breached password detection into our auth management system. Enabling real time compromised password detection and mitigation in FusionAuth couldn't be easier. 
+
+Assuming you've [activated your paid edition with a license key](/docs/v1/tech/reactor), navigate to the Tenant details page, and then to the "Password" tab. Enable "Breached password detection settings" and then choose your options. If you need varied configuration for different applications, use the multi-tenant feature to create separate tenants and configure them individually.
+
+{% include _image.liquid src="/assets/img/blogs/breached-password-detection/enable-breached-password-check.png" alt="Enabling breached password detection." class="img-fluid" figure=false %}
+
+This tab is also where you can specify various password settings. The defaults are fine, but feel free to change the rules to meet your needs, whether requiring certain types of characters, setting an expiration time, or mandating a minimum length. Just don't make me hop on one foot, please.
+
+When configuring breached password detection, there are three levels of compromised password matching. You can match:
+
+* on the password alone
+* on the password plus username, email address or email sub-address
+* on the password and an exact match on username or email address, or commonly used passwords like "password". 
+
+While you know your security requirements best, we recommend matching on password alone, as this provides the most protection. 
+
+Matching rules apply to all passwords provided whenever a user is created. This includes:
+
+* when they register themselves, should self-registration be enabled
+* when they are created via the administrative user interface 
+* when a user is created via the [User APIs](/docs/v1/tech/apis/users)
+
+In all cases, a provided password will be checked against a large and ever-growing database of compromised credentials maintained by FusionAuth. In addition, any time a password is changed by a user or administrator, the database will also be consulted. 
+
+The other configuration option is what, if any, action to take on user login. Should a password be checked when someone signs in? One option is to simply not check for compromised passwords during the sign in event. This is a good choice if you enable breached password detection before you have any production users. In that case, all passwords will have already been vetted during user creation. However, if a password is compromised after an account is created, a user won't be notified until they change their password.
```

Comment:
Since the breached set continues to expand, the check at login is important if the company using FusionAuth wants to ensure that users are continually protected. Since their current password might be safe today, but breached tomorrow, this check is still generally a good idea. You cover all of that here, but it might be best to reword this avoid suggesting that if they enabled breach password detection early on they are safer or that it is okay. The only reason they should disable login checks is if they are concerned about performance. Usually that isn't much of an issue and FusionAuth optimizes that to reduce the calls to Reactor when the data set hasn't changed.

Reply:
Reworked to make it clear about the tradeoff and to recommend against it.